R/modules.R

```diff
@@ -103,6 +103,36 @@ append_module <- function(modules, module) {
   modules
 }
 
+#' Extract a `landing_module` from list of `modules`
+#' @param modules `teal_modules`
+#' @keywords internal
+#' @return `landing_module`
+extract_landing <- function(modules) {
```

Comment:
I think we need a general extract/drop function which accept any logical condition (or function).
```suggestion
module_extract_by <- function(modules, by) {
```

So that, module_extract_by could be used like this
```
module_extract_by(modules, class(module) == "teal_module_landing")
module_extract_by(modules, class(module) == "teal_module_previewer")

# or 
module_extract_by(modules, function(module) class(module) == "teal_module_landing")
```


Same function will be able to do following:
```r
module_extract_by(modules, module$lange == "my label")
```

I motivate this because we have some similar recursive inline function to find if any module is of class previewer.

is_any_previewer

https://github.com/insightsengineering/teal/blob/a71f06cb04b6101d9421643b92567ffde5d964cc/R/module_teal.R#L229

Reply:
so maybe `is_any_module_class <- function(modules, class)`?

Reply:
How about `extract_module(modules, class)`? Operating on classes does not create the need to reserve labels (currently we have two of those).

Reply:
Simpler solution might be usage of `rapply`. We probably need to remove remove `$children` slot from `teal_modules` and simply include `teal_module` in the list. Then recursive operation could be simpler

Reply:
makes sense, will try to work on it!

Reply:
I tried, this was easier.

Reply:
I think my suggestion goes to far. I think it could be done in #889.

Since extract_landing and drop_landing is used once you can skip creating a function and write inline code in teal::init. We can argue about implementation in #889 how to manage nested module.

Reply:
I'm leaning towards `extract_module(modules, class)` if solution with `rapply` is too much

Reply:
just cant see yet if it's gonna be easier to do during the deprecation of `modules()` or now on this PR?

Reply:
Made it more generic with @chlebowa suggestion in here 69d5c714cd18f48f87eb22108a0a2c8ad3691236
Also substituted `is_reporter_previewer` with new `extract/drop_module` functions,

Reply:
The problem with `rapply` is you need `Filter` and `do.call` at the end so `rapply` calls would still have to be processed. I like it better if the action is self-contained.